File: BibliotecaUFABC/app/src/main/java/com/nintersoft/bibliotecaufabc/notification/SyncNotificationDisplay.java
package com.nintersoft.bibliotecaufabc.notification;

import android.app.AlarmManager;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import androidx.core.content.ContextCompat;

import com.nintersoft.bibliotecaufabc.synchronization.SyncService;
import com.nintersoft.bibliotecaufabc.utilities.GlobalConstants;
import com.nintersoft.bibliotecaufabc.utilities.GlobalFunctions;
import com.nintersoft.bibliotecaufabc.R;

public class SyncNotificationDisplay extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context.getApplicationContext());
        long nextAlarm = System.currentTimeMillis()
                - preferences.getLong(context.getString(R.string.key_synchronization_schedule), 0);
        if (nextAlarm > AlarmManager.INTERVAL_DAY * GlobalConstants.SYNC_REMINDER_NOTIFICATION_INTERVAL) {
            GlobalFunctions.createSyncNotification(context,
                    R.string.notification_sync_title,
                    R.string.notification_sync_message,
                    intent.getIntExtra(GlobalConstants.NOTIFICATION_ID, GlobalConstants.SYNC_NOTIFICATION_REMINDER_ID));
            ContextCompat.startForegroundService(context, new Intent(context, SyncService.class));
        }
    }
}

File: BibliotecaUFABC/app/src/main/java/com/nintersoft/bibliotecaufabc/notifications/SyncNotificationDisplay.kt
package com.nintersoft.bibliotecaufabc.notifications

import android.content.Context
import androidx.preference.PreferenceManager
import androidx.work.Worker
import androidx.work.WorkerParameters
import com.nintersoft.bibliotecaufabc.R
import com.nintersoft.bibliotecaufabc.global.Constants
import com.nintersoft.bibliotecaufabc.global.Functions
import java.util.concurrent.TimeUnit

class SyncNotificationDisplay(private val context: Context, workerParams: WorkerParameters) :
    Worker(context, workerParams) {

    override fun doWork(): Result {
        PreferenceManager.getDefaultSharedPreferences(context).also { prefs ->
            val nextAlarm = System.currentTimeMillis() - prefs.getLong(context.
                getString(R.string.key_synchronization_schedule), 0)
            if (nextAlarm >= TimeUnit.DAYS.toMillis(Constants.SYNC_REMINDER_NOTIFICATION_INTERVAL)){
                Functions.createSyncNotification(
                    R.string.notification_sync_title,
                    R.string.notification_sync_message, Constants.SYNC_NOTIFICATION_REMINDER_ID)
            }
        }
        return Result.success()
    }

}